sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriterV2.scala

```diff
@@ -153,6 +153,7 @@ final class DataFrameWriterV2[T] private[sql](table: String, ds: Dataset[T])
    */
   @throws(classOf[NoSuchTableException])
   def append(): Unit = {
+    assertNoTempView("append")
     val append = loadTable(catalog, identifier) match {
```

Comment:
I'm thinking of a more aggressive refactor, which creates an unresolved logical plan here and leaves the table/temp view lookup to the analyzer.

For example, here we can just create `AppendData.byName(UnresolvedRelation(...))`.

By doing this, we can make the framework more clear: the API layer should just create logical plans, other works should be done by the analyzer and query planner.

Reply:
I agree the suggestion is promising, but I'm not sure I understand all spots I need to modify. I could try to find places where `InsertIntoStatement(UnresolvedRelation(...))` is resolved, but AppendData and InsertIntoStatement are not 100% same. In addition, looks like it should be done for each operation.

I'll take a look at this soon. If you can fix it easily (and you'd like to) please go ahead and I can learn from your PR.

Reply:
done in https://github.com/apache/spark/pull/29970